### ISSUE TITLE
arguments detection for jsdoc / javascript has a false positive

### ISSUE THREAD

**Issue by Raynos** [microsoft/TypeScript#47004]
Typescript infers that if you touch the `arguments` value inside a function that it has extra arguments as `...args: any[]`

However if you call a method with an object and one of the fields is called `arguments: []` then it thinks you are using the `arguments` variable inside of the function, then it adds `...args: any[]` 

![image](https://user-images.githubusercontent.com/479538/144620034-a11be375-75ab-4018-b853-874e8c3e7693.png)

https://www.typescriptlang.org/play?filetype=js#code/LAKAxg9gdgzgLgAhgT3gUwLYILwIN5JpQAmAFBAJT4C+C1ooAZhBAEYCGATqQOTRo8KAbgYgA9ACoJoBBIQABAA5d2WPPE4BLKAHNa-GRLFMArlDBxN0BMzZcE5KGip4ZSVHEwA6GETKuQBCCEdkVNAC4EHlsOTh4AGjcgrh0TDCI4GEiAbR4ACzQAG0KIHgBdN2oKUGogA

```js
const system = { send(o) {} }

foobar('one');

/**
 * @param {string} one
 */
function foobar (one) {
  system.send({
    api: 'foobar',
    arguments: ['hello']
  })
}
```

**Comment by jimmywarting** [comment]
I have a similar problem

but i'm actually using `arguments.length` to just simply validating/asserting the number of arguments. This only occur in `.js` (not in `.ts`)

javascript: <img src="https://i.stack.imgur.com/eIAQo.png" width="400" ><br><br>
typescript: <img src="https://i.stack.imgur.com/ufI87.png" width="400" >


I think the better option would be to not add `...args[]: any` at all if you use arguments...
the new normal is that you instead use spread when you actually mean to support any number of arguments

```js
class Math {
  /** @param {number} args */
  min (...args) { 
  }
}
```

**Comment by DanielRosenwasser** [comment]
Thanks @a-tarasyuk!

**Comment by jimmywarting** [comment]
if i want to use `arguments` but do not want to have `...args[]: any` then what can I do?
